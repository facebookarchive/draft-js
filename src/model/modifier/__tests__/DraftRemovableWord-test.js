/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall draft_js
 */

'use strict';

expect.addSnapshotSerializer(require('NonASCIIStringSnapshotSerializer'));

const DraftRemovableWord = require('DraftRemovableWord');

let forward;
let backward;

const accents = 'th\u00e9 f\u00e0bregas';
const arabic = '\u0637\u0638\u0639 \u063A\u063B\u063C';
const cyrillic = '\u0430\u0448\u043e\u043a \u0431\u0414\u0416\u0426';
const english = 'the animals';
const halfWidthHangul = '\uffa3\uffa6\uffb0 \uffa1\uffa2\uffb2';
const japanese = '\u4f1a\u8b70\u4e2d \u30cf\u30c3\u30b7\u30e5';
const korean = '\ud2b8\uc704\ud130 \ud2b8\uc704\ud130';
const withNumbers = 'f14 tomcat';

beforeEach(() => {
  jest.resetModules();
  forward = DraftRemovableWord.getForward;
  backward = DraftRemovableWord.getBackward;
});

test('must identify words looking forward', () => {
  expect(forward(english)).toMatchSnapshot();
  expect(forward(accents)).toMatchSnapshot();
  expect(forward(arabic)).toMatchSnapshot();
  expect(forward(japanese)).toMatchSnapshot();
  expect(forward(korean)).toMatchSnapshot();
  expect(forward(halfWidthHangul)).toMatchSnapshot();
  expect(forward(cyrillic)).toMatchSnapshot();
  expect(forward(withNumbers)).toMatchSnapshot();
});

test('must identify words with apostrophes looking forward', () => {
  expect(forward("you're correct.")).toMatchSnapshot();
});

test('must identify words with curly quotes looking forward', () => {
  expect(forward('you\u2019re correct.')).toMatchSnapshot();
});

test('must identify punctuation with apostrophes', () => {
  expect(forward("'hello'")).toMatchSnapshot();
  expect(backward("'hello'")).toMatchSnapshot();
  expect(forward('\u2018hello\u2019')).toMatchSnapshot();
  expect(backward('\u2018hello\u2019')).toMatchSnapshot();

  expect(forward("('hello')")).toMatchSnapshot();
  expect(backward("('hello')")).toMatchSnapshot();
  expect(forward(".'.")).toMatchSnapshot();
  expect(backward(".'.")).toMatchSnapshot();
});

test('must identify words with underscores', () => {
  expect(forward('under_score hey')).toMatchSnapshot();
  expect(backward('hey under_score')).toMatchSnapshot();
});

test('must identify words led by spaces looking forward', () => {
  expect(forward('  ' + english)).toMatchSnapshot();
  expect(forward('    ' + accents)).toMatchSnapshot();
  expect(forward('      ' + arabic)).toMatchSnapshot();
});

test('must identify words led by punctuation looking forward', () => {
  expect(forward('.' + english)).toMatchSnapshot();
  expect(forward('|' + english)).toMatchSnapshot();
  expect(forward('^' + english)).toMatchSnapshot();
  expect(forward('\u060d\uFD3e\uFD3F' + english)).toMatchSnapshot();
  expect(forward('.. .. ..' + english)).toMatchSnapshot();
});

test('must identify punct/whitespace strings looking forward', () => {
  expect(forward('    ')).toMatchSnapshot();
  expect(forward('\u060d\uFD3e\uFD3F')).toMatchSnapshot();
  expect(forward('. . . . .')).toMatchSnapshot();
  expect(forward('   !!!???   ')).toMatchSnapshot();
});

test('must identify words looking backward', () => {
  expect(backward(english)).toMatchSnapshot();
  expect(backward(accents)).toMatchSnapshot();
  expect(backward(arabic)).toMatchSnapshot();
  expect(backward(japanese)).toMatchSnapshot();
  expect(backward(korean)).toMatchSnapshot();
  expect(backward(halfWidthHangul)).toMatchSnapshot();
  expect(backward(cyrillic)).toMatchSnapshot();
  expect(backward(withNumbers)).toMatchSnapshot();
});

test('must identify words with apostrophes looking backward', () => {
  expect(backward("you don't")).toMatchSnapshot();
});

test('must identify words ended by spaces looking backward', () => {
  expect(backward(english + '  ')).toMatchSnapshot();
  expect(backward(accents + '    ')).toMatchSnapshot();
  expect(backward(arabic + '      ')).toMatchSnapshot();
});

test('must identify words ended by punctuation looking backward', () => {
  expect(backward(english + '.')).toMatchSnapshot();
  expect(backward(english + '|')).toMatchSnapshot();
  expect(backward(english + '^')).toMatchSnapshot();
  expect(backward(english + '\u060d\uFD3e\uFD3F')).toMatchSnapshot();
  expect(backward(english + '.. .. ..')).toMatchSnapshot();
});

test('must identify punct/whitespace strings looking backward', () => {
  expect(backward('    ')).toMatchSnapshot();
  expect(backward('\u060d\uFD3e\uFD3F')).toMatchSnapshot();
  expect(backward('. . . . .')).toMatchSnapshot();
  expect(backward('   !!!???   ')).toMatchSnapshot();
});

test('must identify nothing in an empty string', () => {
  expect(forward('')).toMatchSnapshot();
  expect(backward('')).toMatchSnapshot();
});
