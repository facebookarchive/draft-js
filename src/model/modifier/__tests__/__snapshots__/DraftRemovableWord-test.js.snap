// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must identify nothing in an empty string 1`] = `""`;

exports[`must identify nothing in an empty string 2`] = `""`;

exports[`must identify punct/whitespace strings looking backward 1`] = `" "`;

exports[`must identify punct/whitespace strings looking backward 2`] = `"\\ufd3f"`;

exports[`must identify punct/whitespace strings looking backward 3`] = `"."`;

exports[`must identify punct/whitespace strings looking backward 4`] = `" "`;

exports[`must identify punct/whitespace strings looking forward 1`] = `" "`;

exports[`must identify punct/whitespace strings looking forward 2`] = `"\\u060d"`;

exports[`must identify punct/whitespace strings looking forward 3`] = `"."`;

exports[`must identify punct/whitespace strings looking forward 4`] = `" "`;

exports[`must identify punctuation with apostrophes 1`] = `"'"`;

exports[`must identify punctuation with apostrophes 2`] = `"hello'"`;

exports[`must identify punctuation with apostrophes 3`] = `"\\u2018"`;

exports[`must identify punctuation with apostrophes 4`] = `"hello\\u2019"`;

exports[`must identify punctuation with apostrophes 5`] = `"("`;

exports[`must identify punctuation with apostrophes 6`] = `")"`;

exports[`must identify punctuation with apostrophes 7`] = `"."`;

exports[`must identify punctuation with apostrophes 8`] = `"."`;

exports[`must identify words ended by punctuation looking backward 1`] = `"."`;

exports[`must identify words ended by punctuation looking backward 2`] = `"|"`;

exports[`must identify words ended by punctuation looking backward 3`] = `"^"`;

exports[`must identify words ended by punctuation looking backward 4`] = `"\\ufd3f"`;

exports[`must identify words ended by punctuation looking backward 5`] = `"."`;

exports[`must identify words ended by spaces looking backward 1`] = `" "`;

exports[`must identify words ended by spaces looking backward 2`] = `" "`;

exports[`must identify words ended by spaces looking backward 3`] = `" "`;

exports[`must identify words led by punctuation looking forward 1`] = `"."`;

exports[`must identify words led by punctuation looking forward 2`] = `"|"`;

exports[`must identify words led by punctuation looking forward 3`] = `"^"`;

exports[`must identify words led by punctuation looking forward 4`] = `"\\u060d"`;

exports[`must identify words led by punctuation looking forward 5`] = `"."`;

exports[`must identify words led by spaces looking forward 1`] = `" "`;

exports[`must identify words led by spaces looking forward 2`] = `" "`;

exports[`must identify words led by spaces looking forward 3`] = `" "`;

exports[`must identify words looking backward 1`] = `"animals"`;

exports[`must identify words looking backward 2`] = `"f\\u00e0bregas"`;

exports[`must identify words looking backward 3`] = `"\\u063a\\u063b\\u063c"`;

exports[`must identify words looking backward 4`] = `"\\u30cf\\u30c3\\u30b7\\u30e5"`;

exports[`must identify words looking backward 5`] = `"\\ud2b8\\uc704\\ud130"`;

exports[`must identify words looking backward 6`] = `"\\uffa1\\uffa2\\uffb2"`;

exports[`must identify words looking backward 7`] = `"\\u0431\\u0414\\u0416\\u0426"`;

exports[`must identify words looking backward 8`] = `"tomcat"`;

exports[`must identify words looking forward 1`] = `"the"`;

exports[`must identify words looking forward 2`] = `"th\\u00e9"`;

exports[`must identify words looking forward 3`] = `"\\u0637\\u0638\\u0639"`;

exports[`must identify words looking forward 4`] = `"\\u4f1a\\u8b70\\u4e2d"`;

exports[`must identify words looking forward 5`] = `"\\ud2b8\\uc704\\ud130"`;

exports[`must identify words looking forward 6`] = `"\\uffa3\\uffa6\\uffb0"`;

exports[`must identify words looking forward 7`] = `"\\u0430\\u0448\\u043e\\u043a"`;

exports[`must identify words looking forward 8`] = `"f14"`;

exports[`must identify words with apostrophes looking backward 1`] = `"don't"`;

exports[`must identify words with apostrophes looking forward 1`] = `"you're"`;

exports[`must identify words with curly quotes looking forward 1`] = `"you\\u2019re"`;

exports[`must identify words with underscores 1`] = `"under_score"`;

exports[`must identify words with underscores 2`] = `"under_score"`;
